src/panels/config/integrations/integration-panels/zwave_js/capability-controls/zwave_js-capability-control-door-lock.ts

```diff
@@ -0,0 +1,326 @@
+        ? html`
+            <div class="row">
+              <ha-textfield
+                type="number"
+                .label=${this.hass.localize(
+                  "ui.panel.config.zwave_js.node_installer.capability_controls.door_lock.hold_release_time"
+                )}
+                .value=${this._configuration?.holdAndReleaseTime?.toString() ??
+                ""}
+                @change=${this._numberChanged}
+                key="holdAndReleaseTime"
+              >
+              </ha-textfield>
+            </div>
+          `
+        : nothing}
+
+      <div class="actions">
+        <ha-progress-button @click=${this._saveConfig}>
+          ${this.hass.localize("ui.common.save")}
+        </ha-progress-button>
+      </div>
+    `;
+  }
+
+  protected firstUpdated() {
+    this._loadConfiguration();
+    this._loadCapabilities();
+  }
+
+  private async _loadConfiguration() {
+    try {
+      const config = await invokeZWaveCCApi<DoorLockConfiguration | null>(
+        this.hass,
+        this.device.id,
+        this.command_class,
+        this.endpoint,
+        "getConfiguration",
+        [],
+        true
+      );
+      this._configuration = config ?? {
+        // The server can return null but I think a real device will always have a configuration
+        operationType: 1,
+        outsideHandlesCanOpenDoorConfiguration: [false, false, false, false],
+        insideHandlesCanOpenDoorConfiguration: [false, false, false, false],
+      };
+    } catch (err) {
+      this._error = extractApiErrorMessage(err);
+    }
+  }
+
+  private async _loadCapabilities() {
+    try {
+      const capabilities = await invokeZWaveCCApi<DoorLockCapabilities | null>(
+        this.hass,
+        this.device.id,
+        this.command_class,
+        this.endpoint,
+        "getCapabilities",
+        [],
+        true
+      );
+      this._capabilities = capabilities ?? {
+        supportedOperationTypes: [1, 2],
+      };
+    } catch (err) {
+      this._error = extractApiErrorMessage(err);
+    }
+  }
+
+  private _operationTypeChanged(ev: CustomEvent) {
+    const target = ev.target as HTMLSelectElement;
+    const newType = parseInt(target.value);
+    if (this._configuration) {
+      this._configuration = {
+        ...this._configuration,
+        operationType: newType,
+        // Clear the timeout configuration if switching away from timed operation
+        lockTimeoutConfiguration:
+          newType === 2
+            ? this._configuration.lockTimeoutConfiguration
+            : undefined,
+      };
+    }
+  }
+
+  private _booleanChanged(ev: CustomEvent) {
+    const target = ev.target as HaSwitch;
+    const key = ev.detail.key;
+    if (this._configuration) {
+      this._configuration = {
+        ...this._configuration,
+        [key]: target.checked,
+      };
+    }
+  }
+
+  private _numberChanged(ev: CustomEvent) {
+    const target = ev.target as HTMLInputElement;
+    const key = ev.detail.key;
```

Comment:
This line causes an error because `ev.detail` is undefined.
Should this be 
```suggestion
    const key = target.getAttribute("key");
```
instead?